src/Sentry/Laravel/Tracing/Storage/TracingFilesystem.php

```diff
@@ -0,0 +1,153 @@
+<?php
+
+namespace Sentry\Laravel\Tracing\Storage;
+
+use Illuminate\Contracts\Filesystem\Cloud as CloudFilesystemContract;
+use Illuminate\Contracts\Filesystem\Filesystem;
+use Illuminate\Filesystem\FilesystemAdapter;
+use Sentry\Tracing\SpanContext;
+use function Sentry\trace;
+
+class TracingFilesystem implements CloudFilesystemContract
+{
+    protected $filesystem;
+
+    public function __construct(Filesystem $filesystem)
+    {
+        $this->filesystem = $filesystem;
+    }
+
+    /**
+     * @template TResult
+     * @param array<string, mixed> $data
+     * @param callable(): TResult $originalCall
+     * @return TResult
+     */
+    protected function withTracing(string $method, array $data, callable $originalCall)
+    {
+        $context = new SpanContext();
+        $context->setOp("storage.{$method}");
+        $context->setData($data);
+
+        return trace($originalCall, $context);
+    }
+
+    public function assertExists($path, $content = null)
+    {
+        return $this->withTracing('assertExists', ['path' => $path], function () use ($path, $content) {
+            // TODO assertExists only present on FilesystemAdapter
+            return $this->filesystem->assertExists($path, $content);
+        });
+    }
+
+    public function url($path)
+    {
+        return $this->withTracing('url', ['path' => $path], function () use ($path) {
+            return $this->filesystem->url($path);
+        });
+    }
+
+    public function exists($path)
+    {
+        // TODO: Implement exists() method.
+    }
+
+    public function get($path)
+    {
+        // TODO: Implement get() method.
+    }
+
+    public function readStream($path)
+    {
+        // TODO: Implement readStream() method.
+    }
+
+    public function put($path, $contents, $options = [])
+    {
+        return $this->withTracing('put', ['path' => $path], function () use ($path, $contents, $options) {
```

Comment:
I thought we better exclude `$contents` from this, but I am not sure. What if it is a very large file, does the SDK handle that?

We could add `$options` though.

If we decide to just add all parameters in any case, the implementation can be significantly simplified.

Reply:
This has the potential to increase the payload size significantly, which might lead to dropped events that are hard to debug in the PHP SDKs. Additional concerns are PII and sensitive data. The best way to achieve this would be the use of attachments, but this is something we have to implement in an upcoming major version of the PHP SDK first.

For now, the file path is hopefully enough to debug issues.

Reply:
Thanks for the feedback, I documented the reasoning and implemented all methods accordingly.

Reply:
I actually found usability without description pretty poor, so I just added the JSON encoded data. Does not look to bad: 
![image](https://github.com/getsentry/sentry-laravel/assets/12158000/bd0ad824-7db3-4cf0-8c97-d7564d005945)